cloud/pkg/router/utils/path.go

```diff
@@ -0,0 +1,69 @@
+package utils
+
+import (
+	"regexp"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	ParamRegex = "{[-A-Za-z0-9+&@#%?=~_|!:,.;]+}"
+	pathRegex  = "[-A-Za-z0-9+&@#%?=~_|!:,.;]+"
+)
+
+// URLToURLRegex return url regex and replace {} in url
+func URLToURLRegex(url string) string {
+	paramRegex := regexp.MustCompile(ParamRegex)
+	params := paramRegex.FindAllString(url, -1)
+	for _, param := range params {
+		url = strings.Replace(url, param, pathRegex, -1)
+	}
+	url = url + "/{0,1}"
+	return url
+}
+
+// isMatch return true if the request uri math rule using regex
+func IsMatch(reg, path string) bool {
+	match, err := regexp.MatchString(URLToURLRegex(reg), path)
+	if err != nil {
+		klog.Errorf("failed to validate res %s and reqPath %s, err: %v", reg, path, err)
+		return false
+	}
+	if match {
+		return true
+	}
+	return false
+}
+
+// RuleContains return true if rule 1 contains rule 2
+func RuleContains(rulePath []string, rule2Path []string) bool {
```

Comment:
1. more clear definition: `func RuleContains(r, subrule []string)bool` see for example https://golang.org/pkg/strings/#Contains
2. why rule is `[]string`?

Reply:
3. just list some exmples

Reply:
done